### DISCUSSION TITLE
When will bugs be fixed?

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dotnet/maui#21168] [General]
Does anyone know what is happening with bug fixes? Things have gone a little quiet since the .NET 8 update and it looks like the bug fixing has come to a complete standstill.

I have critical bugs which are causing user backlash, and I literally have no idea what to tell my customer. These are my top 2:

https://github.com/dotnet/maui/issues/19383 
https://github.com/dotnet/maui/issues/19214

The keyboard issue is actually unbelievable - after years of development this basic function still doesn't work. Someone half heartedly looked at it a few months ago but seems to have stalled yet again.

**Comment by davidortinau** [comment]
For visibility on bug fixing, see the project board here https://github.com/orgs/dotnet/projects/194

We've shipped 3 service releases since .NET 8 GA, most recently today promoting this release to NuGet https://github.com/dotnet/maui/releases/tag/8.0.10. 

#19214 it looks similar to the content inset behavior fixed in this PR which is part of 8.0.10. https://github.com/dotnet/maui/pull/20562

**Reply by jonmdev** [reply]
@davidortinau, any thoughts on this (in my opinion very critical) bug?

https://github.com/dotnet/maui/issues/17884

We cannot currently even display text in Android correctly without letters or words getting clipped/wrapped out of view inappropriately. My app builds have missing words and letters randomly in Android which cannot work.

There has been zero activity on it since I reported it 5-6 months ago and it continues to be an issue in current .NET 8.

My theory is it is due to improper Android label measurement as per this report: https://github.com/dotnet/maui/issues/18328 

It is a mess for an incredibly basic function. What are your thoughts?

**Reply by williambuchanan2** [reply]
Ok, so it looks like the iOS keyboard fix was released yesterday so I will check that out. 

What about https://github.com/dotnet/maui/issues/19383 ? How do I find where it is in the schedule to be fixed? It is a critical issue on iPhone 14 onwards where CollectionView and Images are used, so if I can see roughly when someone will look at it I will have something to report to my customer.

**Reply by williambuchanan2** [reply]
So it seems the keyboard fix hasn't worked (https://github.com/dotnet/maui/issues/19214)...

**Reply by williambuchanan2** [reply]
@davidortinau looks like the keyboard fix which went live isn't related to the bug listed above, so both the above bugs are still major issues.

**Comment by AlleSchonWeg** [comment]
I completely agree. I know the team is doing what ever is possible. If XF retire we are in a bad situation. We can't ship a migrated app which are not stable to our customers. The customers don't know the technical differences and have no understanding why something stopped working.

**Comment by developer9969** [comment]
we are affected badly by the keyboard issues and overlapping

**Comment by sanamhub** [comment]
seeing the activities, is maui dying? i was gonna give it a shot lol

**Reply by williambuchanan2** [reply]
Hard to tell at this stage.... I can't see how it can continue as it is, so unless something drastically changes then it probably will die.

**Reply by jonmdev** [reply]
@williambuchanan2 and @sanamhub I don't think it will die. I see new fixes and workarounds constantly being posted and worked on. I am happy to see that I personally only have one crippling bug now in my way that has had no attention from the team (the Android label sizing error bug as noted above).

There are some efficiency issues I will need to try to figure out also and hopefully improve with.

Compared to the alternatives, I think it is a good system overall. Perhaps I am biased though, as I am using only Border, Label, Editor, Entry, Image, AbsoluteLayout, Shadow, and MediaElement with all my own systems outside those basic elements. So bugs in the more esoteric functions aren't hitting me as much.

Remember that Android, iOS, and Windows are all mature operating systems with virtually no changes in their systems over time at this stage.

ie. Once MAUI is functioning well it will be very easy to maintain. They are making incremental progress towards that. 

From my investigations, there has been no better native system I could find for iOS/Android/Windows App development. 

I suspect and hope it will all be functioning well enough in 6-12 months. If so, this will still be torture for those people needing to quit on Xamarin in the next few months. There is no likely way it will be entirely production ready by that short of a time frame.

Certainly I still believe the team is vastly underfunded by Microsoft. If a clear bug report can be made on a simple component (eg. Label, Editor) with <100 lines code to reproduce, this should not take >6 months for them to even consider fixing. And fixes themselves should not be in progress >6 months without being completed. 

Probably they ought to have twice as many people as they do.

**Reply by sanamhub** [reply]
@jonmdev let's hope for the best 🤞

**Reply by FM1973** [reply]
Give Maui hybrid a try if you don´t need native controls.

**Reply by mohachouch** [reply]
@jonmdev  'Perhaps I am biased though, as I am using only Border, Label, Editor, Entry, Image, AbsoluteLayout, Shadow, and MediaElement with all my own systems outside those basic éléments'. 

I think that's exactly the problem, you don't trust the MAUI component, you had to create your own. But this comes at a cost because you will have to take care of the maintenance of its components.

We made exactly the same mistake, when starting Xamarin Forms, we also had problems with basic components, so we created a lot of renderers. When switching to MAUI we estimated the cost of migrating all of its 'homemade' components.

It was more expensive to migrate all of its components to MAUI than to go to migrate to other framework.

**Reply by jonmdev** [reply]
Not exactly @mohachouch . I have worked in other systems before and I simply prefer to manage more complex systems myself. It allows more control and more unique interfaces and designs. Even if Maui was working perfectly I would do the same. I did the same in prior systems. I am not using any xaml either and never would. So I am not exactly a typical user.

**Reply by williambuchanan2** [reply]
I'll just follow up by saying that I hope it doesn't die because I think if they can fix up the problems it will be a great platform. I have found the general day to day development to be quite nice (if I blank out all the problems we are having), and I think the general principle of Maui is well thought through. They have just messed up real bad with the implementation and have lost a lot of good will in the process. I hope that doesn't cause their demise.

They said a year ago that they were adding to the team and speeding up bug fixes, but from what I can see this doesn't appear to have happened, so it just keeps adding to the frustration.
